plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/VarbinaryDecoder.java

```diff
@@ -45,13 +46,16 @@ else if (value instanceof String) {
         }
     }
 
-    public static byte[] toBytes(String stringValue)
+    public static byte[] toBytes(Object value)
```

Comment:
`ByteArray` argument is used only from `PinotSegmentPageSource`. Revert this change and inline `Hex.decodeHex`.

Reply:
Good suggestion. I will change according your advice.